guide/interactions/registering-slash-commands.md

```diff
@@ -0,0 +1,120 @@
+
+```js
+client.once('ready', async () => {
+	const data = {
+		name: 'ping',
+		description: 'Replies with Pong!',
+	};
+
+	const command = await client.application?.commands.create(data);
+	console.log(command);
+});
+```
+
+That's it! You've successfully created your first global application command! Let's move on to guild specific commands.
+
+
+## Guild commands
+
+Guild specific application commands are only available in the guild they have been created in, as such we'll be using `GuildApplicationCommandManager#create()` to create them:
+
+```js{7}
+client.once('ready', async () => {
+	const data = {
+		name: 'ping',
+		description: 'Replies with Pong!',
+	};
+
+	const command = await client.guilds.cache.get('id')?.commands.create(data);
+	console.log(command);
+});
+```
+
+Excellent! Now you've learned how to register guild specific application commands. We're not done yet, there's still a few more topics to cover, keep reading!
+
+
+## Bulk-update commands
+
+If you for example deploy your application commands when starting your application, you may want to update all commands and their changes at once. You can do this by passing an array of `ApplicationCommandData` objects to the `set()` method on either of the managers you've been introduced to above: 
+
+::: danger
+This will overwrite all existing commands on the application or guild with the new data you provided!
+:::
+
+```js{2-11,13-14}
+client.once('ready', async () => {
+	const data = [
+		{
+			name: 'ping',
+			description: 'Replies with Pong!',
+		},
+		{
+			name: 'pong',
+			description: 'Replies with Ping!',
+		},
+	];
+
+	const commands = await client.application?.commands.set(data);
+	console.log(commands);
+});
+```
+
+Perfect! You have now learned how to bulk-update application commands. 
+
+
+## Options
+
+Application commands can have `options`, think of these options like arguments to a function. You can specify them as seen below:
+
+```js{5-10}
+client.once('ready', async () => {
+	const data = {
+		name: 'echo',
+		description: 'Replies with your input!',
+		options: [{
+    		name: 'input',
+    		type: 'STRING',
+    		description: 'The input which should be echoed back',
+    		required: true,
+		}]
+	};
+
+	const command = await client.application?.commands.create(data);
+	console.log(command);
+});
+```
+
+Notice how we specified `required: true` within the options object? Doing this will make the option a required field and will prevent the user from sending the command without specifying a value for this option!
+
+Now you've learned how to create an application command with `options`. Keep reading, we have one more section to cover!
+
+## Option types
+
+As you've seen in the example of the last section we can specify the `type` of an `ApplicationCommandOption`. Listed below you'll find a list of all possible values you can pass as `ApplicationCommandOptionType`:
```

Comment:
Would it maybe be worth a sentence noting if casing is important here, i.e `'STRING'` vs `'string'`?

Reply:
Fair enough, possibilities are listed below, and if people don't take it from there they'd probably ignore a "random" one line comment anyway.